ci/kind/kind-setup.sh

```diff
@@ -579,6 +583,13 @@ function clean_kind {
           if (( time_difference > 3600 )); then
               echo "The creation of $name happened more than 1 hour ago."
               kind delete cluster --name "$name" || echo "Cluster could not be deleted"
+              # For the cluster that is being deleted, we also delete the VLAN subnets and networks.
+              # Remove the cluster from the list of clusters
+              CLUSTER_NAME="$name"
+              destroy_external_servers
```

Comment:
Overriding the global varible in a loop is error-prone and could cause problem if any later step consumes the global variable.
It should pass the cluster name as a parameter to the functions where it's required.